src/features/mixer.py

```diff
@@ -27,27 +28,35 @@ def __init__(self, iterators, mix_method='sum', shape=None, pos=None, seed=0 ):
                 from the given position on that dimension. If None, slices
                 will be extracted from random positions in that dimension.
 
+        Optional Arguments:
+            seed: (int) random seed to initialize each iterator (default 0)
+            diffseed: (bool) do you want a different random seed for each iterator? 
+
         Iterator yields:
             tuple (mix, *features), total length equal to length of iterators + 1,
                 and mix is the sum of the input features
         '''
         self.iterators = []
+        self.return_key = return_key
         for i, iterator in enumerate(iterators):
             if isinstance(iterator, str):
-                self.iterators.append(Hdf5Iterator(iterator,shape=shape,pos=pos,seed=seed+i))
+                iseed = seed + int(diffseed)*i
+                self.iterators.append(Hdf5Iterator(iterator,shape=shape,pos=pos,seed=iseed, return_key=return_key))
             else:
                 self.iterators.append(iterator)
         self.mix_method = mix_method
 
     def __next__(self):
         next_example = next(zip(*self.iterators))
         logger = logging.getLogger(__name__)
-        logger.debug("dtype of first dataset: {}".format(
-            next_example[0].dtype))
-        logger.debug(len(next_example))
-        logger.debug(','.join([str(x.dtype) for x in next_example if isinstance(x, np.ndarray)]))
-        if self.mix_method == 'sum':
-            mixed_example = np.sum(np.array(next_example), axis=0)
+        # logger.debug("dtype of first dataset: {}".format( next_example[0].dtype))
+        # logger.debug(len(next_example))
+        # logger.debug(','.join([str(x.dtype) for x in next_example if isinstance(x, np.ndarray)]))
+        if self.mix_method == 'sum' or self.mix_method == 'add':
+            if self.return_key:
+                mixed_example = np.array( list( zip( *next_example ) )[1] ).sum(axis=0)
```

Comment:
I assume you are referring to the index [1] in the `next_example`. Since the iterators return a tuple in the form of (key, data), we are summing all the data. This comment has been added into the code, hopefully for better readability.